src/ast.rs

```diff
@@ -72,50 +73,47 @@ impl From<((usize, usize, usize), (usize, usize, usize))> for Span {
 }
 
 /// Represents a name of an entity, such as a type, variable, function, etc.
-#[derive(Debug, PartialEq)]
-pub struct RawIdentifier<'a> {
-    pub name: &'a str,
-}
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawIdentifier(pub String);
 
-pub type Identifier<'a> = Spanned<RawIdentifier<'a>>;
+pub type Identifier = Spanned<RawIdentifier>;
 
-impl<'a> From<ParseInput<'a>> for RawIdentifier<'a> {
+impl<'a> From<ParseInput<'a>> for RawIdentifier {
     fn from(span: ParseInput<'a>) -> Self {
-        RawIdentifier {
-            name: span.fragment(),
-        }
+        RawIdentifier((*span.fragment()).to_owned())
     }
 }
 
-/// A parameter to a function, e.g., `foo: MyType`.
-#[derive(Debug, PartialEq)]
-pub struct RawParameter<'a> {
-    pub name: Identifier<'a>,
-    pub param_type: Identifier<'a>,
+/// Introduces a variable with its name and type, e.g., `foo : MyType`. E.g. used in function
+/// type signature parameter lists, `let` bindings, etc.
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawVariableDeclaration {
+    pub name: Identifier,
+    pub typ: Identifier,
```

Comment:
`typ` ugh

But I don't have a better alternative, esp. if this is true-to-convention.

Reply:
yup, can't use the keyword `type`.